File: baekjoon/dp/1912-consecutive-sum.cpp
//https://www.acmicpc.net/problem/1912

#include <iostream>
#include <algorithm>

using namespace std;
using ll = long long;

ll arr[100001];
ll d[100001];

int main() {
	int n;
	ll res = -1e15;
	cin >> n;
	for (int i = 1; i <= n; i++) {
		cin >> arr[i];
	}

	for (int i = 1; i <= n; i++) {
		d[i] = max(d[i - 1] + arr[i], arr[i]);
		res = max(res, d[i]);
	}
	cout << res << endl;
	return 0;
}

File: baekjoon/dp/1912/1912-consecutive-sum.py
n = int(input())
a = [int(i) for i in input().split()]

d = []
d.append(a[0])
res = d[0]

for i in range(1, len(a)):
    d.append(max(d[i - 1] + a[i], a[i]))
    res = max(d[i], res)
    print(res)

print(res)